src/main/java/de/tum/in/www1/artemis/service/exam/ExamImportService.java

```diff
@@ -138,33 +134,93 @@ public List<ExerciseGroup> importExerciseGroupsWithExercisesToExistingExam(List<
                 }
             });
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Checks if a project with the same key and name already exists on VCS / CI. The number of such occurrences is counted
+     *
+     * @param programmingExercises that are checked for an existing project
+     * @param courseShortName      the short name of the course the exercises will be imported into
+     * @return the number of exercises that need to be renamed in the client
+     */
+    private AtomicInteger checkForExistingProjectAndRemoveTitleShortName(List<Exercise> programmingExercises, String courseShortName) {
+        // Count how many programming exercises have conflicts with VCS / CI due to the project with the same key / name already existing
+        AtomicInteger numberOfInvalidProgrammingExercises = new AtomicInteger();
+
+        // Iterate over all programming exercises
+        programmingExercises.forEach(exercise -> {
+            // Method to check, if the project already exists.
+            boolean projectExists = programmingExerciseService.preCheckProjectExistsOnVCSOrCI((ProgrammingExercise) exercise, courseShortName);
+            if (projectExists) {
+                // If the project already exists the short name and title are removed. It has to be set in the client again
+                exercise.setShortName("");
+                exercise.setTitle("");
+                numberOfInvalidProgrammingExercises.getAndIncrement();
+            }
         });
+        return numberOfInvalidProgrammingExercises;
+    }
+
+    /**
+     * Checks that all programming exercises of the given exercise group have a unique title and short name.
+     * Additionally, checks if an exercise with the same project key or name already exists on the VCS / CI.
+     * In case of an invalid configuration, the exam is sent back to the client with the title / short name removed, wherever a new one must be chosen
+     *
+     * @param exerciseGroups  the list of all exercises (not only programming) to be checked
+     * @param courseShortName the short name of the course the exercises will be imported into
+     * @throws ExamConfigurationException in case of duplicated titles / short names or if one or more programming exercise project keys are not unique
+     */
+    private void preCheckProgrammingExercisesForTitleAndShortNameUniqueness(List<ExerciseGroup> exerciseGroups, String courseShortName) {
+        List<Exercise> programmingExercises = exerciseGroups.stream().flatMap(group -> group.getExercises().stream())
+                .filter(exercise -> exercise.getExerciseType() == ExerciseType.PROGRAMMING).toList();
+
+        // check for duplicated titles
+        boolean duplicatedTitles = checkForAndRemoveDuplicatedTitlesAndShortNames(programmingExercises, true);
+        if (duplicatedTitles) {
+            throw new ExamConfigurationException(exerciseGroups, 0, "duplicatedProgrammingExerciseTitle");
```

Comment:
The number is not really used, this is why I set it to 0. The error message only says "Multiple" instead of the concrete number because I think it is more confusing than informative to give a number, since the user cannot be sure how the duplicates are counted. Number of all duplicates together or how many different duplicates there are? Still I have to give a number to be able to reuse the `ExamConfigurationException` that was already used before the PR